contracts/token/WithdrawTokens.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+
+/**
+ * @title Making a token withdrawable by the owner to a specified onlyOwner setable address
+ *
+ * @dev
+ */
+abstract contract WithdrawTokens is OwnableUpgradeable {
+    address public beneficiary;
+
+    event BeneficiaryUpdated(address beneficiary);
+
+    constructor(address _owner) {
+        transferOwnership(_owner);
+    }
+
+    function setBeneficiary(address _beneficiary) public onlyOwner {
+        require(_beneficiary != address(0), "WithdrawTokens/beneficiary zero");
+        beneficiary = _beneficiary;
+        emit BeneficiaryUpdated(_beneficiary);
+    }
+
+    function withdrawERC20Tokens(IERC20Upgradeable token, uint256 amount)
+        public
+        onlyOwner
+    {
+        // can we safely make this non-onlyOwner?
+        // approvals?
+
+        token.transfer(beneficiary, amount); // use safe transfer lib?
```

Comment:
issue: check the return vale

Copy the logic from safe transfer with a custom revert message.

Reply:
this was actually stale code - slither didnt like it, so I had pushed a fix.

Reply:
seeing that it wasn't just that return value it disliked but also https://github.com/crytic/slither/wiki/Detector-Documentation#public-function-that-could-be-declared-external